Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -612,7 +645,59 @@ void ReflectometryReductionOne::exec() {
   }
 
   IvsQ = this->toIvsQ(IvsLam, correctDetectorPositions, theta, isPointDetector);
-
+  double momentumTransferMinimum = getProperty("MomentumTransferMinimum");
+  double momentumTransferStep = getProperty("MomentumTransferStep");
+  double momentumTransferMaximum = getProperty("MomentumTransferMaximum");
+  MantidVec QParams;
+  if (!isDefault("MomentumTransferMinimum") &&
+      !isDefault("MomentumTransferStep") &&
+      !isDefault("MomentumTransferMaximum")) {
+    QParams.push_back(momentumTransferMinimum);
+    QParams.push_back(-momentumTransferStep);
+    QParams.push_back(momentumTransferMaximum);
+  } else {
+    momentumTransferMinimum = calculateQ(IvsLam->readX(0).back(), theta.get());
+    momentumTransferMaximum = calculateQ(IvsLam->readX(0).front(), theta.get());
+    if (isDefault("MomentumTransferStep")) {
+      // if the DQQ is not given for this run.
+      // we will use CalculateResoltion to produce this value
+      // for us.
+      IAlgorithm_sptr calcResAlg =
+          AlgorithmManager::Instance().create("CalculateResolution");
+      calcResAlg->setProperty("Workspace", runWS);
+      calcResAlg->setProperty("TwoTheta", theta.get());
+      calcResAlg->execute();
+      if (!calcResAlg->isExecuted())
+        throw std::runtime_error("CalculateResolution failed. Please manually "
+                                 "enter a value in the dQ/Q column.");
+      momentumTransferStep = calcResAlg->getProperty("Resolution");
+    }
+    QParams.push_back(momentumTransferMinimum);
+    QParams.push_back(-momentumTransferStep);
+    QParams.push_back(momentumTransferMaximum);
+  }
+  IAlgorithm_sptr algRebin = this->createChildAlgorithm("Rebin");
```

Comment:
Yes, it is mandatory now. There are two main reasons:
1) It was mandatory in the interface, i.e. our old `ReflTableViewPresenter` would always apply this `Rebin` step.
2) Jos and Max mentioned that they want the interface to do exactly what the algorithm does, they don't want the interface to do extra steps. They also mentioned that they usually want their `IvsQ` workspace to be binned logarithmically.